src/js/views/postTweet.js

```diff
@@ -1,48 +1,58 @@
 import React from "react";
 import TwitterService from '../services/twitterService.js';
+import PubSubListener from './pubSubListener.js';
 
 class PostTweetUI extends React.Component {
   constructor(props){
     super(props);
     this.state = {
-      numChars: 0,
-      postTweetButtonDisabled: true,
-      userTweetText: '',
+      postTweetText: props.postTweetText,
       postStatus: 'pending',
     }
   }
 
+  componentWillReceiveProps(nextProps){
+    if(nextProps) {
+      if(nextProps.postTweetText !== this.state.postTweetText) {
+        this.setState({
+          postTweetText: nextProps.postTweetText,
+          postStatus: 'pending',
+        });
+      }
+    }
+  }
+
   preparePost(event) {
-    let userTweetText = event.target.value;
-    let numChars = userTweetText.length;
     this.setState({
-      numChars: numChars,
-      postTweetButtonDisabled: !(numChars > 0 && numChars <= 280),
-      userTweetText: userTweetText,
+      postTweetText: event.target.value,
       postStatus: 'pending',
     });
   }
 
   postAndGetResponse() {
     let twitterService = new TwitterService;
-    let postResponse;
-    twitterService.postTweet(this.state.userTweetText).then(res => {
-      res ? this.setState({postStatus: 'success'}) : this.setState({postStatus: 'fail'});
+    this.props.isReply ?
+      twitterService.replyTweet(this.state.postTweetText, this.props.tweet.id).then(res => {
```

Comment:
check to ensure id exists before even allowing the button click
this.props.tweet && this.props.tweet.id || 0